### ISSUE TITLE
Add `Memory::grow_to_minimum` or add more utilities to `Pages`

### ISSUE THREAD

**Issue by tomaka** [paritytech/wasmi#659]
In smoldot, I would like to ensure that an existing `Memory` object is *at least* N pages.

Right now, the only way to grow a `Memory` is to call `grow`, which increases the size of the memory by the given number.

In theory, you'd think that you can simply something like `memory.grow(minimum - memory.current_pages())`

However, the `Pages` struct doesn't have any way neither to retrieve the actual value as a number, nor to subtract a number of pages.
In other words, there is no `impl From<Pages> for u16` nor `impl Pages { fn checked_sub(&self, &Pages) { ... } }` nor `impl Sub<Pages> for Pages`.

**Comment by Robbepop** [comment]
> In other words, there is no `impl From<Pages> for u16` nor `impl Pages { fn checked_sub(&self, &Pages) { ... } }` nor `impl Sub<Pages> for Pages`.

Should be fairly simple add if that's all what is needed.

**Comment by Robbepop** [comment]
While it is a bit ugly there already exists an `impl From<Pages> for u32` which together with `Pages::new(u32)` constructor could be used to implement your use case.

**Comment by tomaka** [comment]
> @tomaka While it is a bit ugly there already exists an impl From<Pages> for u32 which together with Pages::new(u32) constructor could be used to implement your use case. What is strange is that this From impl does not show up in the crate docs. thinking

Ah thank you, good to know
